### ISSUE TITLE
Add `ARCH=loongarch`

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/continuous-integration2#599]
Requires TuxMake/TuxSuite support first: https://gitlab.com/Linaro/tuxmake/-/issues/205

Once it is supported there, the following configurations need to be disabled to build at the time of writing this:

- https://github.com/ClangBuiltLinux/linux/issues/1883
```
CONFIG_CRASH_DUMP=n
CONFIG_RELOCATABLE=n
```
- https://github.com/ClangBuiltLinux/linux/issues/1884
```
CONFIG_MODULES=n
```

Additionally, [the support for LoongArch in `ld.lld`](https://reviews.llvm.org/D138135) needs to be merged to use full `LLVM=1`, otherwise we may be able to use GNU ld but I have not tested that yet.

**Comment by nathanchance** [comment]
> Requires TuxMake/TuxSuite support first: https://gitlab.com/Linaro/tuxmake/-/issues/205

https://gitlab.com/Linaro/tuxmake/-/merge_requests/325

That is only one piece, we need `ld.lld` support merged into LLVM main to allow us to use `clang-nightly`, as Debian does not ship `loongarch64-linux-gnu` binutils or GCC.

**Comment by xen0n** [comment]
FYI: the LLD LoongArch port [has been upstreamed](https://github.com/llvm/llvm-project/commit/6084ee742064cf8121444febee342133b0cb7abd) just in time for the LLVM 17 release. We will continue the work on enabling the other bits ;-)

**Comment by nathanchance** [comment]
I have opened https://gitlab.com/Linaro/tuxsuite/-/issues/194 as a prerequisite.

**Comment by nickdesaulniers** [comment]
commit 5a31ed4678e0 ("LoongArch: Mark Clang LTO as working")

so we should add LTO builds, too

**Comment by nickdesaulniers** [comment]
What's the latest on this?

**Comment by nathanchance** [comment]
This is blocked on https://gitlab.com/Linaro/tuxmake/-/issues/207 now.
